aws-cpp-sdk-core/source/monitoring/MonitoringManager.cpp

```diff
@@ -93,7 +94,7 @@ namespace Aws
             {
                 return;
             }
-            s_monitors = Aws::MakeUnique<Monitors>(MonitoringTag);
+            s_monitors = Aws::MakeUniqueWithDeleter<Monitors>(MonitoringTag, MonitorsDeleter(&s_monitors));
```

Comment:
and maybe reduce boilerplate here so it looks like:

`Aws::MakeStaticUnique(&s_monitors, MonitoringTag);`


and that helper function's definition is something like:
```
template<class T, typename ... Args>
StaticUniquePtr<T> MakeStaticUnique(T* pointer, Args&&... args)
```

it's been a while since I've ~done~ fought templates ... so no idea if that's all accurate

Reply:
The problem with make_static is that C++ standard semantic/logic enforces that unique_ptrs allocated via std::make_unique to not have a custom deleter.... so "WithDeleter" is a very important addition to this function mocking the std::make_unique and violating the initial std::unique_ptr design.

However, an overload with a default deleter provided is feasible here.